plone/app/vocabularies/terms.py

```diff
@@ -25,7 +25,10 @@ def safe_simpleterm_from_value(value):
     - token need cleaned up: Vocabulary term tokens *must* be 7 bit values
     - anything for display has to be cleaned up, titles *must* be unicode
     """
-    return SimpleTerm(value, b2a_qp(safe_encode(value)), safe_unicode(value))
+    return SimpleTerm(value,
+                      b2a_base64(safe_encode(value), newline=False),
+                      safe_unicode(value)
+                      )
```

Comment:
This blews up the tokens size a lot - and so the size of large vocabularies!
Also, the newline parameter was introduced in Python 3.6 and wont work in 2.7.
```suggestion
    if isinstance(value, six.text_type):
        try:
            token = value.decode('ascii')
        except UnicodeDecodeError:
            token = b2a_base64(safe_encode(value)).strip()
    else:
        token = b2a_base64(safe_encode(value)).strip()
    return SimpleTerm(value, token, safe_unicode(value))
```
This could be improved to handle non-text/buffer values, but this exceeds the intend of the PR.

Reply:
token = value.encode('ascii')

Reply:
I understand.   For unicode values, it makes sense to base64 encode, but for ascii encodable tokens it does not make sense - and increases the size of the token.   I agree.